docs/user/instrument-android-crashes-example.md

```diff
@@ -0,0 +1,255 @@
+defined.  First, we need to add an import line:
+
+```Kotlin
+import org.mozilla.gleancrashexample.GleanMetrics.Pings
+```
+
+Then we need to register our custom ping by calling `Glean.registerPings(Pings)`.  In the `MainActivity.onCreate()` function, add
+the following line:
+
+```Kotlin
+Glean.registerPings(Pings)
+```
+
+This registers the custom ping with Glean so that it knows about it and can manage the storage and other important details of
+it like sending it when we tell it to.
+
+### Instrument The App To Record The Event
+
+Next, we need to make the `MainActivity` handle uncaught exceptions, so we extend the class definition by adding
+`Thread.UncaughtExceptionHandler` as an inherited class like this:
+
+```Kotlin
+class MainActivity : AppCompatActivity(), Thread.UncaughtExceptionHandler {
+    ...
+}
+```
+
+In order to be a responsible `Thread.UncaughtExceptionHandler`, we need to implement an override for the `uncaughtException()`
+function.  Somewhere in your `MainActivity` class, add the following override:
+
+```Kotlin
+override fun uncaughtException(p0: Thread, p1: Throwable) {
+    Crash.exception.record(
+        mapOf(
+            Crash.exceptionKeys.cause to p1.cause!!.toString(),
+            Crash.exceptionKeys.message to p1.message!!)
+    )
+    Pings.crash.send()
+}
+```
+
+To explain what’s happening here, first we are recording to the `Crash.exception` metric we created.  If you recall, the
+category of the metric was `crash` and the name was `exception` so we access it by calling `record()` on the `Crash.exception`
+object that was created by the magic of Glean.  You can also see where we are passing in the extra information for the cause
+and the message which will get packaged up and sent along with our ping when the second action of `Pings.crash.send()` is
+called.  Basically this forces our ping to be sent immediately after the recording of the event.
+
+Finally, we need to register our `MainActivity` as the default uncaught exception handler by adding the following line to the
+`onCreate()` function, after `Glean.initialize(this)`:
+
+```Kotlin
+Thread.setDefaultUncaughtExceptionHandler(this)
+```
+
+### Make A Way To Crash The App
+
+That's pretty much it!  Now we just need to make a way to crash our app with an uncaught exception.  The "Basic Activity"
+template has a floating action button called `fab` that can be used, and it's set up in the `onCreate()` of the `MainActivity`.  
+We can change the click listener to make it look like this in order to make it throw an exception:
+
+```Kotlin
+fab.setOnClickListener {
+    // Go boom!
+    throw NullPointerException()
+}
+```
+
+Now run the application and click the floating action button and you should have just crashed the app while recording and
+sending crash telemetry!  I should mention, this information didn't really get recorded by anything, as there is additional
```

Comment:
As part of our documentation I think it would be better to point this out more formally.
Something along the lines of:

In order for the data to be properly recorded by our ingestion pipeline, additional work is required (link to it, I think we have that documented somewhere?)
This might be more automatic in the future.